packages/engine/Specs/Core/TaskProcessorSpec.js

```diff
@@ -60,6 +65,52 @@ describe("Core/TaskProcessor", function () {
     );
   });
 
+  it("when workers loaded via module ID and it is cross-origin, loads worker with appropriate shim", async function () {
+    // Setup a cross origin BASE_URL
+    const oldCESIUM_BASE_URL = window.CESIUM_BASE_URL;
+    window.CESIUM_BASE_URL = "http://test.com/source/";
+    buildModuleUrl._clearBaseResource();
+
+    const blobSpy = spyOn(window, "Blob");
+
+    // Provide just the module ID, as is prevalent in the codebase
+    taskProcessor = new TaskProcessor("transferTypedArrayTest");
+
+    try {
+      await taskProcessor.scheduleTask();
+    } catch {
+      // We expect this to throw as we cannot setup a true cross-origin base in tests
```

Comment:
Would it possible to call dummy code instead such that this doesn't throw? For example, could the blob spy return a blobbed dummy script?

Reply:
I've added the ability to skip running the task for testing purposes in 20d967149de28ac39748c319b883bdb71dfb0d9a

Mocking out a dummy worker script with necessary message handlers so that it doesn't hang got a bit bulky and I think this results in a cleaner test, but happy to revisit if you'd prefer to avoid this approach